feature/#342-elevation-update:

joesnellpdx opened the pull request:
Update the shadow values per the following specs: https://share.goabstract.com/9c3b0991-2207-4136-9543-9a8ac98656e9

- add new elevation tokens for '08' opacity and `spirit-elevation-secondary`
- update mixin to show secondary level of elevation
- update mixin doc block

**NOTES**
* Did **NOT** update manually crafted elevations that do not use the mixin (i.e. button)
* Not sure if implementation utilization of tokens is _best_ solution: looking for feedback
* I wanted to make sure existing usage of mixin on this project and potentially in others will not break - so had to keep the params the same. 
* In sketch - elevation 5 default had a different 'elevation' than the rest (1-4 default, and 1-5 blue) - I assume that is incorrect in sketch and keep it uniform across all secondary elevations (`0 0 4px 0`). If this requires adjustment, will make mixin and manual utilizations much more complicated.

I added many reviewers for feedback on implementation and notes above

View: `/sink-pages/visual-style/elevations.html` for results

jamesmelzer commented:
@kevinmpowell Can you take a look at the code? I can't speak to optimization.

kevinmpowell commented:
@joesnellpdx I'm thinking about ways to minimize the complexity and encapsulate it. Here's a thought of one way to do that by hard-coding these shadow values into tokens. This will simplify the mixin and make it less likely that someone will combine incorrect elevation variables together:

```
elevation :
  0 : &elevation-dimensions-0 "0 0 0 0"
  1 : &elevation-dimensions-1 "0 2px 4px 0"
  2 : &elevation-dimensions-2 "0 4px 8px 0"
  3 : &elevation-dimensions-3 "0 8px 16px 0"
  4 : &elevation-dimensions-4 "0 12px 24px 0"
  5 : &elevation-dimensions-5 "0 24px 48px 0"
  secondary: &elevation-dimensions-secondary "0 0 4px 0"
  shadow:
    default:
      0 : "!{*elevation-dimensions-0} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
      1 : "!{*elevation-dimensions-1} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
      2 : "!{*elevation-dimensions-2} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
      3 : "!{*elevation-dimensions-3} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
      4 : "!{*elevation-dimensions-4} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
      5 : "!{*elevation-dimensions-5} rgba(!{*color-black}, .16), !{elevation-dimensions-secondary} rgba(!{*color-black}, .08)"
    blue:
      0 : "!{*elevation-dimensions-0} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
      1 : "!{*elevation-dimensions-1} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
      2 : "!{*elevation-dimensions-2} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
      3 : "!{*elevation-dimensions-3} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
      4 : "!{*elevation-dimensions-4} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
      5 : "!{*elevation-dimensions-5} rgba(!{*color-blue-100}, .16), !{elevation-dimensions-secondary} rgba(!{*color-blue-100}, .08)"
```

I would suggest removing the `elevation-color` key from `tokens.yaml` as well.

This puts more complexity into the tokens file, but I think that's the place where these atomic decisions should be captured. I believe this would still work in the case of `button.scss`, where an additional shadow is required, for example:

instead of:
```
  box-shadow: $spirit-elevation-1 $spirit-elevation-color-blue, inset 0 0 0 $spirit-border-width-button $spirit-interactive-color-primary;
```

you'd have:
```
  box-shadow: $spirit-elevation-shadow-blue-1, inset 0 0 0 $spirit-border-width-button $spirit-interactive-color-primary;
```

joesnellpdx commented:
@kevinmpowell 

Please review. SEE: https://github.com/JDRF/spirit/pull/345/commits/3c75845d782cb64137afd5b882b9ff588aebb178

- Simplified mixin per your recommendation
- Added elevation shadow tokens - for default, blue, success, info, danger and transparent (all used in sink and .org)
- Kept elevation color as it is used for backgrounds when some components are used (i.e. modal)
- Modified all uses of old variables and old params for mixins (many of these also used in .org)

Once approved - need to make .org ticket to update mixins / vars when release consumed
